tests/platform_tests/mellanox/util.py

```diff
@@ -0,0 +1,329 @@
+            expected_threshold_data_keys_and_pattern.update(expected_module_threshold_values_keys_and_value_pattern)
+
+            logging.info("Check if MonitorData's keys exist and the corresponding value format is correct")
+            check_dom_monitor_key_and_data_format(expected_monitor_data_keys_and_pattern,
+                                                  sfp_eeprom_info["MonitorData"])
+
+            logging.info("check if ThresholdData's keys exist and the corresponding value format is correct")
+            check_dom_monitor_key_and_data_format(expected_threshold_data_keys_and_pattern,
+                                                  sfp_eeprom_info["ThresholdData"])
+
+        else:
+            if "ChannelThresholdValues" not in excluded_keys:
+                logging.info(
+                    "Check if ChannelThresholdValues' keys exist and the corresponding value format is correct")
+                check_dom_monitor_key_and_data_format(expected_channel_threshold_values_keys_and_value_pattern,
+                                                      sfp_eeprom_info["ChannelThresholdValues"])
+
+            if "ModuleThresholdValues" not in excluded_keys:
+                logging.info("Check if ModuleThresholdValues' keys exist and the corresponding format is correct")
+                check_dom_monitor_key_and_data_format(expected_module_threshold_values_keys_and_value_pattern,
+                                                      sfp_eeprom_info["ModuleThresholdValues"])
+
+            logging.info("Check if ChannelMonitorValues's value format is correct")
+            for k, v in sfp_eeprom_info["ChannelMonitorValues"].items():
+                pattern = pattern_power if "Power" in k else pattern_bias
+                assert re.match(pattern, v), "Value of {}:{} format is not correct. pattern is {}".format(k, v, pattern)
+
+            logging.info("Check ModuleMonitorValues keys exist and the corresponding value format is correct")
+            check_dom_monitor_key_and_data_format(expected_module_monitor_values_keys_and_pattern,
+                                                  sfp_eeprom_info["ModuleMonitorValues"])
+
+
+def check_dom_monitor_key_and_data_format(expected_keys_and_pattern_dict, dom_monitor_data):
+    for key, pattern in expected_keys_and_pattern_dict.items():
+        assert key in dom_monitor_data, "Key {} doesn't exist in {}".format(key, dom_monitor_data)
+        assert re.match(pattern, dom_monitor_data[key]), \
+            "Value of {}:{} format is not correct. pattern is {}".format(
+                key, dom_monitor_data[key], pattern)
+
+
+def is_support_dom(duthost, port_index):
+    """
+    This method is to check if cable support dom
+    """
+    bulk_status_str = get_transceiver_bulk_status(duthost, port_index)
+    bulk_status_str = bulk_status_str.replace('-inf', '\'-inf\'')
+    bulk_status_dict = eval(bulk_status_str)
```

Comment:
suggest use `literal_eval` instead of `eval`

Reply:
ok